File: Python_Scripting_for_Computational_Science_Third_Edition/perl/datatrans2.pl
: # *-*-perl-*-*
  eval 'exec perl -w -S  $0 ${1+"$@"}' 
    if 0;  # if running under some shell

die "Usage: $0 infilename outfilename\n" if $#ARGV < 1;
($infilename, $outfilename) = @ARGV;

open(INFILE,  "<$infilename") 
    or die "unsuccessful opening of $infilename; $!\n";
@lines = <INFILE>; # read file into array of lines
close(INFILE);

# go through each line and split line into x and y columns:
@x = (); @y = ();   # store data pairs in two arrays x and y
for $line (@lines) {
    ($xval, $yval) = split(' ', $line);
    push(@x, $xval);  push(@y, $yval);
}

open(OUTFILE, ">$outfilename")
    or die "unsuccessful opening of $outfilename; $!\n";

for ($i = 0; $i <= $#x; $i++) {
    $fy = myfunc($y[$i]);  # transform y value
    printf(OUTFILE "%g  %12.5e\n", $x[$i], $fy);
}
close(OUTFILE);

sub myfunc {
    my ($y) = @_;
    if ($y >= 0.0) { return $y**5.0*exp(-$y); }
    else           { return 0.0; }
}

File: Python_Scripting_for_Computational_Science_Third_Edition/py/intro/datatrans2.py
#!/usr/bin/env python
import sys

try:
    infilename  = sys.argv[1]
    outfilename = sys.argv[2]
except:
    print "Usage:",sys.argv[0], "infile outfile"; sys.exit(1)
    
ifile = open(infilename, 'r')  # open file for reading
lines = ifile.readlines()      # read file into list of lines
ifile.close()

# go through each line and split line into x and y columns:
x = []; y = []   # store data pairs in two lists x and y
for line in lines:
    xval, yval = line.split()
    x.append(float(xval)); y.append(float(yval))

from math import exp
def myfunc(y):
    return (y**5*exp(-y) if y >= 0 else 0.0)

ofile = open(outfilename, 'w') # open file for writing
for i in range(len(x)):
    fy = myfunc(y[i])  # transform y value
    ofile.write('%g  %12.5e\n' % (x[i],fy))
ofile.close()
